### PR TITLE
Validate `Logon` message before acting upon `ResetMsgSeqNum` flag

### PR DESCRIPTION
**Pull request by harinda05**
Fixes https://github.com/quickfix-j/quickfixj/issues/474 - 

verify() method call moved further up in nextLogon() to prevent seqNumber resets over invalid logins.

### CODE REVIEW

**Comment by chrjohn** [comment]
@harinda05 
Thanks for the PR. Could you please base this against master?
Thanks in advance

**Comment by harinda05** [comment]
> [chrjohn](/chrjohn)

Changed the base branch to master

File: quickfixj-core/src/main/java/quickfix/Session.java [hunk 1]
```diff
@@ -1771,12 +1771,14 @@ private boolean verify(Message msg, boolean checkTooHigh, boolean checkTooLow)
                 return false;
             }
 
-            if (checkTooHigh && isTargetTooHigh(msgSeqNum)) {
-                doTargetTooHigh(msg);
-                return false;
-            } else if (checkTooLow && isTargetTooLow(msgSeqNum)) {
-                doTargetTooLow(msg);
-                return false;
+            if(!state.isResetReceived()){
```
Comment (chrjohn) [inline quickfixj-core/src/main/java/quickfix/Session.java:1774]: So if you receive a ResetSeqNum flag you reset in either case, without checking sequence numbers of the Logon message. This somewhat contradicts what you write in #482 

> When a user sends a Logon message to the system for a user, with ResetSeqNumFlag = Y and MsgSeqNum !=1 we expect the system to break the TCP/IP connection after sending a Logout message due to the MsgSeqNum being invalid.

So is #482 no longer a problem and can be closed?

Comment (harinda05) [reply quickfixj-core/src/main/java/quickfix/Session.java:1774]: I think for the time being #482 can be ignored. Better to revisit that scenario in my opinion after more investigation

**Pull request merged**